src/validation.ts

```diff
@@ -0,0 +1,137 @@
+import { Client, cacheExchange, fetchExchange } from '@urql/core';
+import {
+  type DscvrFramesRequest,
+  type DscvrValidationResponse,
+  dscvrClientProtocolPrefix,
+  DscvrUntrustedData,
+} from './types';
+import { DEFAULT_DSCVR_API_URL } from './default';
+
+type ValidationType = string | number | bigint | null | undefined;
+interface ValidatedQueryResult {
+  user: {
+    id: string;
+  };
+  content?: {
+    id: string;
+  };
+  state: string | null;
+  inputText: string | null;
+  timestamp: string;
+  url: string;
+  buttonIndex: number;
+}
+
+const validationQuery = `#graphql
+    query validate($message: String!) {
+      unpackFrameMessage(message: $message) {
+        user {
+            id
+        },
+        content {
+            id
+        }
+        buttonIndex,
+        state,
+        url,
+        timestamp,
+        inputText,
+      }
+    }
+`;
+
+export const validateClientProtocol = (clientProtocol: string) => {
+  return clientProtocol.startsWith(dscvrClientProtocolPrefix);
+};
+
+const validateUntrustedDataValue = (
+  validated: ValidationType,
+  untrusted: ValidationType
+): boolean => {
+  if (!validated && !untrusted) {
+    return true;
```

Comment:
Why is it `true` in this case? Feels like this method should simply be ```return validated && untrusted && validated === untrusted```

Reply:
Oh I get it, if it's not present in the untrusted or validated then it's also ok. I would simply this to:

```
return (!validated && !untrusted) || (validated === untrusted)
```

Reply:
Fixed